### PR TITLE
authenticate-from-token

### PR DESCRIPTION
**Pull request by ohadbitt**
authenticate-from-token

### CODE REVIEW

File: data/src/test/java/com/microsoft/azure/kusto/data/ConnectionStringBuilderTest.java [hunk 1]
```diff
@@ -118,4 +118,22 @@ void createWithAadApplicationCertificate() throws CertificateException, Operator
...
+                () -> ConnectionStringBuilder
+                        .createWithAadUserTokenAuthentication(null, "token"));
+        Assertions.assertThrows(IllegalArgumentException.class,
+                () -> ConnectionStringBuilder
+                        .createWithAadUserTokenAuthentication("","token"));
+        Assertions.assertThrows(IllegalArgumentException.class,
+                () -> ConnectionStringBuilder
+                        .createWithAadUserTokenAuthentication("resource.uri", null));
+        Assertions.assertThrows(IllegalArgumentException.class,
+                () -> ConnectionStringBuilder
```
Comment (toshetah) [inline data/src/test/java/com/microsoft/azure/kusto/data/ConnectionStringBuilderTest.java]: Can you please add a positive test, where we initiate the CSB with valid token, and we get it when we try?

Comment (ohadbitt) [reply data/src/test/java/com/microsoft/azure/kusto/data/ConnectionStringBuilderTest.java]: we dont do it anywhere else but sure i can

File: data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java [hunk 2]
```diff
@@ -52,6 +54,9 @@ public AadAuthenticationHelper(@NotNull ConnectionStringBuilder csb) throws URIS
             privateKey = csb.getPrivateKey();
             clientCredential = new ClientCredential(csb.getApplicationClientId(), null);
             authenticationType = AuthenticationType.AAD_APPLICATION_CERTIFICATE;
+        } else if (StringUtils.isNotEmpty(csb.getUserToken())){
```
Comment (tamirkamara) [inline data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java]: I think this should be blank (not empty)

File: data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java [hunk 3]
```diff
@@ -29,12 +29,14 @@
     private X509Certificate x509Certificate;
     private PrivateKey privateKey;
     private AuthenticationType authenticationType;
+    private String userToken;
```
Comment (tamirkamara) [inline data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java]: I think this should be renamed across all files to **accessToken**

Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java]: i used the variable name from our .Net library

Comment (tamirkamara) [reply data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java]: Unless I'm missing something I think that name isn't right on the other implementation.
Look at AadAuthenticationHelper.acquireAccessToken - userToken is actually returned and used as access token.
You can confirm with @toshetah to make sure.

Comment (toshetah) [reply data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java]: The userToken is defined in the connection string builder.
In the AadHelpers it is accessToken.

File: data/src/test/java/com/microsoft/azure/kusto/data/ConnectionStringBuilderTest.java (hunk 1 above)
Comment (toshetah) [reply data/src/test/java/com/microsoft/azure/kusto/data/ConnectionStringBuilderTest.java]: I meant more like Assertions.Equels(ConnectionStringBuilder           .createWithAadAccessTokenAuthentication("resource.uri","token").getAccessToken(), "token")

**Pull request merged**